Source/EmbeddedBoundary/WarpXInitEB.cpp

```diff
@@ -163,26 +162,23 @@ WarpX::ComputeFaceAreas () {
     auto const &area_frac = eb_fact.getAreaFrac();
 
     for (amrex::MFIter mfi(flags); mfi.isValid(); ++mfi) {
-        amrex::Box const &box = mfi.validbox();
-        amrex::FabType fab_type = flags[mfi].getType(box);
         for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+            amrex::Box const &box = mfi.tilebox(m_face_areas[maxLevel()][idim]->ixType().toIntVect());
+            amrex::FabType fab_type = flags[mfi].getType(box);
             auto const &face_areas_dim = m_face_areas[maxLevel()][idim]->array(mfi);
             if (fab_type == amrex::FabType::regular) {
                 // every cell in box is all regular
-                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(face_areas_dim).ixType()),
-                                [=](int i, int j, int k) {
+                amrex::ParallelFor(box, [=](int i, int j, int k) {
                     face_areas_dim(i, j, k) = amrex::Real(1.);
                 });
             } else if (fab_type == amrex::FabType::covered) {
                 // every cell in box is all covered
-                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(face_areas_dim).ixType()),
-                                 [=](int i, int j, int k) {
+                amrex::ParallelFor(box, [=](int i, int j, int k) {
                     face_areas_dim(i, j, k) = amrex::Real(0.);
                 });
             } else {
                 auto const &face = area_frac[idim]->const_array(mfi);
-                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(face).ixType()),
-                                [=](int i, int j, int k) {
+                amrex::ParallelFor(box, [=](int i, int j, int k) {
```

Comment:
Shall we add this, too? Cc: @WeiqunZhang 
```suggestion
                amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE (int i, int j, int k) {
```